nimble/host/services/audio/bass/include/services/bass/ble_audio_svc_bass.h

```diff
@@ -0,0 +1,174 @@
+     * event.  The 'type' field indicates which member of the union is valid.
+     */
+    union {
+        /**
+         * Represents Add Source operation.  Valid for the following event
+         * types:
+         *     o BLE_SVC_AUDIO_BASS_CTRL_POINT_EVENT_ADD_SOURCE
+         * Application can accept or reject Add Source operation. If no application callback is set
+         * and free Receive State characteristic exists operation is automatically accepted.
+         * If application callback exists and returns 0 accepted. Otherwise, Operation is rejected.
+         * If operation is accepted by application callback, this callback may select receiver
+         * state to be filled. If application doesnt select characteristic, BASS Server falls back
+         * to searching free one. If none is found, operation is rejected.
+         */
+        struct {
+            /**
+             * Advertiser Address
+             */
+            ble_addr_t adv_addr;
+            uint8_t adv_sid;
+            uint32_t broadcast_id : 24;
+            uint8_t pa_sync;
+            uint16_t pa_interval;
+            uint16_t num_subgroups;
+            struct ble_svc_audio_bass_subgroup
+                subgroups[MYNEWT_VAL(BLE_SVC_AUDIO_BASS_MAX_SUBGROUPS_PER_RECV_STATE)];
+            uint8_t *source_id;
+        } add_source;
+
+        /**newt
+         * Represents Modify Source operation. Valid for the following event
+         * types:
+         *     o BLE_SVC_AUDIO_BASS_CTRL_POINT_EVENT_MODIFY_SOURCE
+         * Application can accept or reject Add Source operation. If no application callback is set
+         * or application callback returns 0 operation is automatically accepted.
+         * If application callback returns non-zero value operation is rejected.
+         */
+        struct {
+            uint8_t source_id;
+            uint8_t pa_sync;
+            uint16_t pa_interval;
+            uint16_t num_subgroups;
+            uint32_t bis_sync[MYNEWT_VAL(BLE_SVC_AUDIO_BASS_MAX_SUBGROUPS_PER_RECV_STATE)];
+        } modify_source;
+
+        /**
+         * Represents Set Broadcast_Code operation. Valid for the following event
+         * types:
+         *     o BLE_SVC_AUDIO_BASS_CTRL_POINT_EVENT_SET_BROADCAST_CODE
+         */
+        struct {
+            uint8_t source_id;
+            uint8_t broadcast_code[BLE_AUDIO_BROADCAST_CODE_SIZE];
+        } set_broadcast_code;
+
+        /**
+         * Represents Remove Source operation. Valid for the following event
+         * types:
+         *     o BLE_SVC_AUDIO_BASS_CTRL_POINT_EVENT_REMOVE_SOURCE
+         */
+        struct {
+            uint8_t source_id;
+        } remove_source;
+    };
+};
+
+typedef int ble_svc_audio_bass_ctrl_point_ev_fn(struct ble_svc_audio_bass_ctrl_point_event *event,
+                                                void *arg);
+
+int
+ble_svc_audio_bass_ctrl_point_cb_set(ble_svc_audio_bass_ctrl_point_ev_fn *cb, void *arg);
```

Comment:
As discussed offline, this should be event generated from generic ble_audio_listener